harness/determined/_data_layer/_data_layer.py

```diff
@@ -10,9 +10,8 @@
 from yogadl import storage, tensorflow
 
 import determined as det
-from determined import horovod
+from determined._generic import DistributedContext
```

Comment:
from determined import _generic
...
_generic.DistributedContext

Prefer to import the final module as a single name (and only import modules, not symbols)